packages/collection-model/lib/model.js

```diff
@@ -250,6 +252,16 @@ const CollectionModel = AmpersandModel.extend(debounceActions(['fetch']), {
         ...collStats,
         ...(collectionInfo && pickCollectionInfo(collectionInfo)),
       });
+      // If the collection is not unprovisioned `is_non_existent` anymore,
+      // let's update the parent database model to reflect the change.
+      // This happens when a user tries to insert first document into a
+      // collection that doesn't exist yet or creates a new collection 
+      // for an unprovisioned database.
+      if (!this.is_non_existent) {
+        getParentByType(this, 'Database').set({
+          is_non_existent: false,
+        });
+      }
```

Comment:
Not sure if this is the right place for this, but if you feel it does not belong here, let me know